lib/beaker-hostgenerator/data.rb

```diff
@@ -29,6 +43,873 @@ def pe_upgrade_family
       'm' => 'master',
     }
 
+    BASE_CONFIG = {
+      'HOSTS' => {},
+      'CONFIG' => {
+        'nfs_server' => 'none',
+        'consoleport' => 443,
+      }
+    }
+
+    def base_host_config
+      {
+        'pe_dir' => pe_dir(pe_version, pe_family),
+        'pe_ver' => pe_version,
+        'pe_upgrade_dir' => pe_dir(pe_upgrade_version, pe_upgrade_family),
+        'pe_upgrade_ver' => pe_upgrade_version,
+      }
+    end
+
+    def osinfo
```

Comment:
Oh my god...I spent so much time dealing with some funky side-effect behavior that manifested between Ruby constants and the `deep_merge`/`deep_merge!` methods that are used to manipulate this map.

I honestly feel like there's a bug in the `deep_merge` module...or it's probably just in Ruby proper.

There are certain combinations of how you define your map and whether you call `deep_merge` or `deep_merge!` that result in buggy behavior, and I've found that defining our map with a function `def osinfo` (instead of const `OSINFO =`) and using `deep_merge!` seems to yield the correct behavior with minimal workaround code.